server/config.js

```diff
@@ -12,22 +15,27 @@ export default function getConfig (dir) {
   return cache.get(dir)
 }
 
-async function loadConfig (dir) {
-  const path = join(dir, 'package.json')
-
-  let data
-  try {
-    data = await readFile(path, 'utf8')
-  } catch (err) {
-    if (err.code === 'ENOENT') {
-      data = '{}'
-    } else {
-      throw err
-    }
+function loadConfig (dir) {
+  const path = join(dir, 'next.config.js')
+  const packagePath = join(dir, 'package.json')
+
+  let userConfig = {}
+  let packageConfig = null
+
+  const userHasConfig = existsSync(path)
+  if (userHasConfig) {
+    const userConfigModule = require(path)
+    userConfig = userConfigModule.default || userConfigModule
   }
 
-  // no try-cache, it must be a valid json
-  const config = JSON.parse(data).next || {}
+  const userHasPackageConfig = existsSync(packagePath)
```

Comment:
Since we are no longer have a `cdn` there's no need to look at the `package.json` or warn the user.